### PR TITLE
Adds Archer

### PR DESCRIPTION
**Pull request by will-rowe**
----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

For members of the Bioconda project, the following command is also available:

<table>
  <tr>
    <td><code>@BiocondaBot please merge</code></td>
    <td>Upload built packages/containers and merge a PR. <br />Someone must approve a PR first! <br />This reduces CI build time by reusing built artifacts.</td>
  </tr>
</table>

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

**Comment by will-rowe** [comment]
@BiocondaBot please add label

File: recipes/archer/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    - {{ compiler('c') }}
+    - {{ compiler('go') }}
+
+test:
+  commands:
+    - archer
+    - archer --version
+
+about:
+  home: https://github.com/will-rowe/version
```
Comment (milot-mirdita) [inline recipes/archer/meta.yaml:25]: shouldn't *version* be *archer*?

File: recipes/archer/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+source:
+  url: https://github.com/will-rowe/archer/archive/refs/tags/v{{ version }}.tar.gz
+  sha256: 34390e947644ae0b239ed4fae7b74a069a11c87e4fd575e19cf3aff2cf89800b
+
+build:
+  number: 0
+
+requirements:
+  build:
+    - {{ compiler('c') }}
```
Comment (milot-mirdita) [inline recipes/archer/meta.yaml:16]: Does it still need a `c` compiler if `CGO_ENABLED=0` is set.

Comment (will-rowe) [reply recipes/archer/meta.yaml:16]: You're right - I can remove that.

File: recipes/archer/meta.yaml (hunk 1 above)
Comment (will-rowe) [reply recipes/archer/meta.yaml:25]: The --version flag was just so I could check the correct version had been built - just sanity checking really. Can remove that extra command call

Comment (milot-mirdita) [reply recipes/archer/meta.yaml:24]: I was referring to the URL, https://github.com/will-rowe/version goes to 404.

Comment (will-rowe) [reply recipes/archer/meta.yaml:24]: sorry - missed that! yes, typo

**Review by dpryan79 — APPROVED** [review]

File: recipes/archer/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    - archer --version
+
+about:
+  home: https://github.com/will-rowe/version
+  license: MIT
+  summary: Pre-process amplicon data before running CLIMB workflows.
+
+extra:
+  skip-lints:
+    - should_be_noarch_generic
```
[the remark this answers is not in the document]
Comment (will-rowe) [reply recipes/archer/meta.yaml:30]: I'll remove it - I had it left in from another recipe I boiler plated this one from and didn't think to get rid of it.

**Comment by milot-mirdita** [comment]
Go seems to be doing something weird with the downloaded modules.

It spams the log with lines like the following:
```
14:29:27 BIOCONDA INFO (OUT) WARNING:conda.gateways.disk.delete:Could not remove or rename /opt/conda/conda-bld/archer_1617891854327/work_moved_archer-0.1.1-he881be0_0_linux-64/gopath/pkg/mod/github.com/golang/protobuf@v1.4.3/.travis.yml.  Please remove this file manually (you may need to reboot to free file handles)
14:29:27 BIOCONDA INFO (OUT) WARNING conda.gateways.disk.delete:unlink_or_rename_to_trash(140): Could not remove or rename /opt/conda/conda-bld/archer_1617891854327/work_moved_archer-0.1.1-he881be0_0_linux-64/gopath/pkg/mod/github.com/golang/protobuf@v1.4.3/.travis.yml.  Please remove this file manually (you may need to reboot to free file handles)
14:29:27 BIOCONDA INFO (OUT) WARNING:conda.gateways.disk.delete:Could not remove or rename /opt/conda/conda-bld/archer_1617891854327/work_moved_archer-0.1.1-he881be0_0_linux-64/gopath/pkg/mod/github.com/golang/protobuf@v1.4.3/AUTHORS.  Please remove this file manually (you may need to reboot to free file handles)
14:29:27 BIOCONDA INFO (OUT) WARNING conda.gateways.disk.delete:unlink_or_rename_to_trash(140): Could not remove or rename /opt/conda/conda-bld/archer_1617891854327/work_moved_archer-0.1.1-he881be0_0_linux-64/gopath/pkg/mod/github.com/golang/protobuf@v1.4.3/AUTHORS.  Please remove this file manually (you may need to reboot to free file handles)
```

There seems to be a parameter `-modcacherw` for `go build` that might fix that spam (see: https://wiki.archlinux.org/index.php/Go_package_guidelines).

**Comment by dpryan79** [comment]
@bioconda-bot please fetch artifacts

**Comment by dpryan79** [comment]
@bioconda-bot please merge

**Pull request merged**